docs/spec.md

```diff
@@ -4001,49 +4001,58 @@ doesn't hold for many popular reductions. E.g. floating-point addition for
 `body` and zero for `init_values` don't actually form a monoid because
 floating-point addition is not associative.
 
-More formally, `results...[j0, ..., jR-1] = reduce(input_slices)` where:
+More formally, `results...[j0, ..., jR-1] = reduce_implicit_convert(reduce(
+input_slices_converted), type(func_outputs(body)...), type(results...)))` where:
 
 * `input_slices = inputs...[j0, ..., :, ..., jR-1]`, where `:` are inserted
   at `dimensions`.
-* `reduce(input_slices) = exec(schedule)` for some binary tree `schedule`
-  where:
+* `input_slices_converted = reduce_implicit_convert(input_slices...,
+  type(inputs...), type(func_inputs(body)...)`.
+* `reduce(input_slices_converted) = exec(schedule)` for some binary tree
+  `schedule` where:
   * `exec(node) = body(exec(node.left), exec(node.right))`.
   * `exec(leaf) = leaf.value`.
 * `schedule` is an implementation-defined full binary tree whose in-order
   traversal consists of:
-  * `input_slices...[index]` values, for all `index` in
-    `index_space(input_slices)` in the ascending lexicographic order of `index`.
-  * Interspersed with an implementation-defined amount of `init_values`
+  * `input_slices_converted...[index]` values, for all `index` in
+    `index_space(input_slices_converted)` in the ascending lexicographic order
+    of `index`.
+  * Interspersed with an implementation-defined amount of
+    `reduce_implicit_convert(init_values..., type(init_values...), type(func_inputs(body)[len(func_inputs(body)//2)]:)...)`
```

Comment:
Should be:

`func_inputs(body)[len(func_inputs(body)//2):])`

(semicolon `:` in the slice operator `[...]`)

Reply:
Actually:

```
func_inputs(body)[len(func_inputs(body))//2:])
```
`len(...)` must not include `//2` either

Reply:
Thanks for pointing point. fixed.